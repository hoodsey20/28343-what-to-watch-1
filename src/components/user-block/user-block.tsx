import * as React from "react";
import * as PropTypes from "prop-types";

export const UserBlock = ({ history, user }) => {
  return (
    <div className="user-block">
      {user ? (
        <div
          className="user-block__avatar"
          onClick={() => {
            history.push(`/mylist`);
          }}
        >
          <img
            src={`https://es31-server.appspot.com${user.avatarUrl}`}
            alt="User avatar"
            width="63"
            height="63"
          />
        </div>
      ) : (
        <a
          onClick={evt => {
            evt.preventDefault();
            history.push(`/login`);
          }}
          href="#"
          className="user-block__link"
        >
          Sign in
        </a>
      )}
    </div>
  );
};

export default UserBlock;

UserBlock.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func
  }),
  user: PropTypes.shape({
    avatarUrl: PropTypes.string
  })
};
