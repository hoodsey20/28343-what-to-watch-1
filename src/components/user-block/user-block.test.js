import React from 'react';
import renderer from 'react-test-renderer';

import {UserBlock} from './user-block.jsx';

const userMock = {"id": 1, "email": `artem.n@cupcakedev.com`, "name": `artem.n`, "avatarUrl": `/wtw/static/avatar/9.jpg`};

it(`UserBlock is rendered correctly`, () => {
  const tree = renderer.create(<UserBlock
    history={{push: jest.fn()}}
    user={userMock}
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
