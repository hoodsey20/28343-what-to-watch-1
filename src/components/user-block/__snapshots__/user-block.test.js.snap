// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock is rendered correctly 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="https://es31-server.appspot.com/wtw/static/avatar/9.jpg"
      width="63"
    />
  </div>
</div>
`;
